sdk/python/kfp/compiler/compiler.py

```diff
@@ -686,6 +686,12 @@ def _create_pipeline_workflow(self, args, pipeline, op_transformers=None, pipeli
 
     if exit_handler:
       workflow['spec']['onExit'] = exit_handler.name
+
+    if pipeline_conf.image_pull_policy != None and pipeline_conf.image_pull_policy in ["Always", "Never", "IfNotPresent"]:
+      for template in workflow["spec"]["templates"]:
+        if "container" in template:
+          template["container"]["imagePullPolicy"] = pipeline_conf.image_pull_policy
```

Comment:
I think that per-task configuration should have priority over pipeline-wide configuration. What do you think?

Reply:
Yes, that makes sense. I have tried to read the source code for [set_image_pull_policy](https://github.com/kubeflow/pipelines/blob/95aec25db437dfadb54124257f0e81ca6f48d14c/sdk/python/kfp/dsl/_container_op.py#L375) and are not really sure how the Container objects are handled. Is it enough to check if it is set and if it is leave it?  I will also think it might make sense to raise a Value error to the user if they try to provide a invalid option like, set_image_pull_policy.

Reply:
I think i got it. Updated with test for this as well.